### PR TITLE
Vault deploy hook (using curl)

### PR DESCRIPTION
**Pull request by pashinin**

### CODE REVIEW

**Comment by pashinin** [comment]
Just added one file. Failed checks are from previous commits as I can see.

File: deploy/vault.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  # So replacing new-lines with "\n" here
+  _ckey=$(sed -z 's/\n/\\n/g' <"$2")
+  _ccert=$(sed -z 's/\n/\\n/g' <"$3")
+  _cca=$(sed -z 's/\n/\\n/g' <"$4")
+  _cfullchain=$(sed -z 's/\n/\\n/g' <"$5")
+
+  URL="$VAULT_ADDR/v1/$VAULT_PREFIX/$_cdomain"
+
+  if [ -n "$FABIO" ]; then
+    curl --silent -H "X-Vault-Token: $VAULT_TOKEN" --request POST --data "{\"cert\": \"$_cfullchain\", \"key\": \"$_ckey\"}" "$URL" || return 1
```
Comment (Neilpang) [inline deploy/vault.sh:54]: don't use `curl`.  please use `_post()` function

**Comment by Neilpang** [comment]
add your usage here: https://github.com/acmesh-official/acme.sh/wiki/deployhooks

**Comment by pashinin** [comment]
Done

**Comment by Cellivar** [comment]
Any ETA on this getting merged? I just got bamboozled by the docs being out of sync with the docker image..

File: deploy/vault.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  _cfullchain="$5"
+
+  _debug _cdomain "$_cdomain"
+  _debug _ckey "$_ckey"
+  _debug _ccert "$_ccert"
+  _debug _cca "$_cca"
+  _debug _cfullchain "$_cfullchain"
+
+  # validate required env vars
+  if [ -z "$VAULT_PREFIX" ]; then
```
Comment (Neilpang) [inline deploy/vault.sh:34]: please use  `_getdeployconf` and `_savedeployconf`  to read and write `VAULT_PREFIX` and `VAULT_ADDR`.

Here is an example.

https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh#L22

Comment (Neilpang) [reply deploy/vault.sh:35]: you need also `_savedeployconf `

**Pull request merged**